grails-datastore-core/src/main/groovy/org/grails/datastore/mapping/config/GormMethodInvokingFactoryBean.groovy

```diff
@@ -0,0 +1,29 @@
+package org.grails.datastore.mapping.config
+
+import groovy.transform.CompileStatic
+import groovy.transform.Internal
+import org.grails.datastore.mapping.core.Datastore
+import org.grails.datastore.mapping.services.Service
+import org.springframework.beans.factory.config.MethodInvokingFactoryBean
+
+/**
+ * Variant of {#link MethodInvokingFactoryBean} which returns the correct data service type instead of {@code java.lang.Object} so the Autowire with type works correctly.
+ */
+@Internal
+@CompileStatic
+class GormMethodInvokingFactoryBean extends MethodInvokingFactoryBean {
```

Comment:
The name for this class doesn't make much sense. It is tied to the concept of loading services. `DatastoreServiceFactoryBean` or something like that